torch/_inductor/lowering.py

```diff
@@ -1556,6 +1556,49 @@ def inner_fn(index):
     )
 
 
+@register_lowering(inductor_prims._bucketize, type_promotion_kind=None)
+def _inductor_bucketize(
+    input: TensorBox,
+    boundaries: TensorBox,
+    *,
+    out_int32: bool = False,
+    right: bool = False,
+):
+    assert len(boundaries.get_size()) == 1
+
+    if input.get_device().type != "cuda" or boundaries.get_device().type != "cuda":
+        return fallback_handler(inductor_prims._bucketize, add_to_fallback_set=False)(
+            input, boundaries, out_int32=out_int32, right=right
+        )
+
+    boundaries_size = boundaries.get_size()[0]
+    boundaries_loader = boundaries.make_loader()
+    device = input.get_device()
+    input_loader = input.make_loader()
+
+    index_dtype = torch.int32 if out_int32 else torch.int64
+    triton_dtype = "tl.int32" if out_int32 else "tl.int64"
+
+    def inner_fn(index):
+        val = input_loader(index)
+        indices = ops.bucketize(
+            val,
+            boundaries.get_name(),
+            ops.index_expr(boundaries_size, index_dtype),
+            triton_dtype,
+            not right,  # ops.bucketize and torch.bucketize have opposite semantics for "right"
```

Comment:
My initial feelings here are we should match the torch semantics rather than the numpy one.  That will at least make our codebase self-consistent.  I don't feel that strongly about that though.